File: 프로그래머스/unrated/181856. 배열 비교하기/배열 비교하기.cpp
#include <string>
#include <vector>
#include <numeric>
#include <math.h>
using namespace std;

int solution(vector<int> arr1, vector<int> arr2) {
    if (arr1.size() == arr2.size())
        return accumulate(arr1.begin(), arr1.end(), 0) == accumulate(arr2.begin(), arr2.end(), 0) ? 0 : accumulate(arr1.begin(), arr1.end(), 0) > accumulate(arr2.begin(), arr2.end(), 0) ? 1 : -1;
    else
        return arr1.size() > arr2.size() ? 1 : -1;
}

File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.java
import java.util.Arrays;

class Solution {
    public int solution(int[] arr1, int[] arr2) {
        return arr1.length == arr2.length ? check(arr1, arr2) : arr1.length > arr2.length ? 1 : -1;

    }


    public static int check(int[] arr1, int[] arr2) {
        int arr1Sum = Arrays.stream(arr1).sum();
        int arr2Sum = Arrays.stream(arr2).sum();
        return arr1Sum == arr2Sum ? 0 : arr1Sum > arr2Sum ? 1 : -1;
    }
    
}
